pyproject.toml

```
@@ -190,10 +190,12 @@ doc = [
     "jinja2<3.1.0", # see https://github.com/readthedocs/readthedocs.org/issues/9037
     "jupyter-client<7.0",
     "nbconvert==5.6.1",
+    "nose==1.3.7",
+    "pytest>=6.1",
     "pyx>=0.15",
     "runnotebook",
     "scipy<=1.7.1", # see https://github.com/yt-project/yt/issues/3966 and https://github.com/scipy/scipy/issues/16602
-    "sphinx==3.1.2",
+    "sphinx==3.5.4",
```

Would be useful to link to an issue here in case anyone sets off on trying to upgrade sphinx

```suggestion
    "sphinx==3.5.4",  # see https://github.com/yt-project/yt/issues/4412
```